Source/LinqToDB.EntityFrameworkCore/LinqToDBContextOptionsBuilder.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+namespace LinqToDB.EntityFrameworkCore
+{
+	using Internal;
+	using Interceptors;
+
+	/// <summary>
+	/// LinqToDB context options builder
+	/// </summary>
+	public class LinqToDBContextOptionsBuilder 
+	{
+		private readonly LinqToDBOptionsExtension _extension;
+
+		/// <summary>
+		/// .ctor
+		/// </summary>
+		/// <param name="optionsBuilder"></param>
+		public LinqToDBContextOptionsBuilder(DbContextOptionsBuilder optionsBuilder)
+		{
+			_extension = optionsBuilder.Options.FindExtension<LinqToDBOptionsExtension>();
+		}
+
+		/// <summary>
+		/// Registers LinqToDb interceptor
+		/// </summary>
+		/// <param name="interceptor">The interceptor instance to register</param>
+		/// <returns></returns>
+		public LinqToDBContextOptionsBuilder AddInterceptor(IInterceptor interceptor)
+		{
+			_extension.Interceptors.Add(interceptor);
+			return this;
+		}
+
+		/// <summary>
+		/// Make the Linq2Db use EF Core registered interceptors
+		/// as long as they also implement Linq2Db interfaces
+		/// </summary>
+		/// <returns></returns>
+		public LinqToDBContextOptionsBuilder UseEfCoreRegisteredInterceptorsIfPossible(bool useEfCoreInterceptors = true)
```

Comment:
Well I thought over this functionality again. I think we will not go with approach. If you need such functionality, you can easily add your own extension method for `LinqToDBContextOptionsBuilder` in your project which will re-register EF Core interceptors. @MaceWindu what do you think?

Reply:
Ok, done. Following your advice I created such sample extension method in tests which does exactly what this functionality did.